core/src/main/java/com/turn/ttorrent/client/Piece.java

```diff
@@ -62,6 +62,18 @@
 	private volatile boolean valid;
 	private int seen;
 	private ByteBuffer data;
+	private static final ThreadLocal<ByteBuffer> validateByteBuffer = new ThreadLocal<ByteBuffer>() {
+		@Override
+		protected ByteBuffer initialValue() {
+			return ByteBuffer.allocate(Torrent.DEFAULT_PIECE_LENGTH);
```

Comment:
You could even be generous and use 4k here (default piece length is 512 bytes I think). A lot of torrents of large size will usually use larger pieces, so if you don't give a little more room here you'll rarely actually benefit from your optimization.